USER_GUIDE.md

```diff
@@ -75,3 +75,100 @@ $client->indices()->delete([
     'index' => $indexName
 ]);
 ```
+
+## ClientBuilder
+
+The `\OpenSearch\ClientBuilder` class is used to create a `\OpenSearch\Client` instance. It provides a fluent interface for configuring the client.
+
+### `setHosts`
+
+This method allows you to set the hosts to use for the client. By default, the `RoundRobinSelector` selector is active, which will select a host from the list of hosts in a round-robin fashion.
+
+```php
+<?php
+    $client = (new \OpenSearch\ClientBuilder())
+        ->setHosts(['https://localhost:9200'])
+        ->build();
+```
+
+### `setSelector`
+
+This method allows you to set the host selection mode to use for the client.
+
+```php
+<?php
+    $client = (new \OpenSearch\ClientBuilder())
+        // Class needs to implement \OpenSearch\ConnectionPool\Selectors\SelectorInterface
+        ->setSelector(new \OpenSearch\ConnectionPool\Selectors\RandomSelector())
+        ->build();
+```
+
+### `setBasicAuthentication`
+
+This method allows you to set the basic authentication credentials to use for the client.
+
+```php
+$client = (new \OpenSearch\ClientBuilder())
+    ->setBasicAuthentication('username', 'password')
+    ->build();
+```
+
+### `setSigV4CredentialProvider`
+
+This method allows you to enable AWS SigV4 authentication for the client. For AWS SigV4 handling you must have to install the AWS SDK.
```

Comment:
must have to -> The AWS SDK is required. 

Add an example of how that is included?